### PR TITLE
user timezone

### PR DESCRIPTION
**Pull request by adnjoo**
#36

### CODE REVIEW

File: app/assets/javascripts/new_expense.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+/**
+* Sets the default value for a datetime input field to the current time in the user's timezone.
+* Retrieves the user's timezone and populates a hidden input field with the timezone value.
+* Updates the datetime input field to display the current time in the user's timezone on page load.
+*/
+document.addEventListener("DOMContentLoaded", function () {
+  var timezoneInput = document.getElementById("user_timezone");
```
Comment (debiday) [inline app/assets/javascripts/new_expense.js:7]: Nit: can you change `var` to `let`?

File: app/assets/config/manifest.js [hunk 2]
```diff
@@ -3,3 +3,4 @@
 //= link_tree ../../javascript .js
 //= link_tree ../../../vendor/javascript .js
 //= link_tree ../builds
+//= link new_expense.js
```
Comment (debiday) [inline app/assets/config/manifest.js:6]: Nit: extra space at the end of the file?

File: app/assets/javascripts/new_expense.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  var datetimeField = document.querySelector("input[type=datetime-local]");
+
+  if (timezoneInput && datetimeField) {
+    var userTimezone = Intl.DateTimeFormat().resolvedOptions().timeZone;
```
Comment (debiday) [inline app/assets/javascripts/new_expense.js:11]: Since this value does not change it can be a `const`

File: app/assets/config/manifest.js (hunk 2 above)
Comment (adnjoo) [reply app/assets/config/manifest.js:6]: hmm weird extra space doesn't show up in GitHub but it's in VSCode

Screenshot 2023-12-17 at 2 55 27 PM

File: app/assets/javascripts/new_expense.js (hunk 3 above)
Comment (adnjoo) [reply app/assets/javascripts/new_expense.js]: ok

**Pull request merged**